gtsam/hybrid/tests/testHybridGaussianFactor.cpp

```diff
@@ -741,6 +741,168 @@ TEST(HybridGaussianFactor, TwoStateModel4) {
   EXPECT(assert_equal(expected, *(bn->at(2)->asDiscrete()), 0.002));
 }
 
+namespace test_direct_factor_graph {
+/**
+ * @brief Create a Factor Graph by directly specifying all
+ * the factors instead of creating conditionals first.
+ * This way we can directly provide the likelihoods and
+ * then perform linearization.
+ *
+ * @param values Initial values to linearize around.
+ * @param means The means of the HybridGaussianFactor components.
+ * @param sigmas The covariances of the HybridGaussianFactor components.
+ * @param m1 The discrete key.
+ * @return HybridGaussianFactorGraph
+ */
+static HybridGaussianFactorGraph CreateFactorGraph(
+    const gtsam::Values &values, const std::vector<double> &means,
+    const std::vector<double> &sigmas, DiscreteKey &m1,
+    double measurement_noise = 1e-3) {
+  auto model0 = noiseModel::Isotropic::Sigma(1, sigmas[0]);
+  auto model1 = noiseModel::Isotropic::Sigma(1, sigmas[1]);
+  auto prior_noise = noiseModel::Isotropic::Sigma(1, measurement_noise);
+
+  auto f0 =
+      std::make_shared<BetweenFactor<double>>(X(0), X(1), means[0], model0)
+          ->linearize(values);
+  auto f1 =
+      std::make_shared<BetweenFactor<double>>(X(0), X(1), means[1], model1)
+          ->linearize(values);
+
+  // Create HybridGaussianFactor
+  std::vector<GaussianFactorValuePair> factors{
+      {f0, ComputeLogNormalizer(model0)}, {f1, ComputeLogNormalizer(model1)}};
```

Comment:
I'm still very shaky about the semantics of this scalar. I asked a question on the previous PR but I don;t think it was answered. If the scalar simply *adds* to the error, then is that after squaring and halving? If so, I think this needs to be better documented, with an example in the .h file even. There is also uncertainty about the sign: is the normalizer multiplying or dividing? I think the former, but again: be very clear, also in the docs of ComputeLogNormalizer. And, very important, it has to be consistent everywhere.

Reply:
I sent you the doc with the math written up there. From the derivation, we are adding the pre-halved version. You are right that it should be better documented in the .h file. :)

`ComputeLogNormalizer` returns $\log(2\pi \Sigma^m)$ instead of $\log(\frac{1}{\sqrt{2\pi \Sigma^m}})$, so we should be dividing (or subtracting since it is in log-space). This is actually convenient since we can pull out $-\frac{1}{2}$ in common from both the factor error and the normalizer term. I'm still blown away by how nicely that math worked out.

Reply:
OK, this answer is problematic to me: when specifying the API for hybrid factor graphs, the normalizer is *just an application*. Ideally, the API in the .h file is documented like:
```
In factor graphs the error function typically returns 0.5*|h(x)-z|^2, i.e., the negative log-likelihood for a Gaussian noise model. In hybrid factor graphs we allow *adding* an arbitrary scalar dependent on the discrete assignment. For example, adding a 70/30 mode probability is supported by providing the scalars $-log(.7)$ and $-log(.3)$. Note that adding a common constant will not make any difference in the optimization, so $-log(70)$ and $-log(30)$ work just as well.
```
Please consider adding the above in HNF and the linear version in HGF. And, not that I am advocating here for **adding** these scalars **terms** to the error. When designing the API for the factors, it does not make sense to tell people that a term will be subtracted: we either work in negative log space (we add) or in probability space (we multiply). I see no need to add subtraction into this lexicon.

Reply:
Happy to chat in person BTW. Invite me.

Reply:
As discussed, I have made all the changes.